src/Components/Blazor/Server/src/MonoDebugProxy/BlazorMonoDebugProxyAppBuilderExtensions.cs

```diff
@@ -237,13 +237,14 @@ private static async Task DebugHome(HttpContext context)
             var underlyingV8Endpoint = tabToDebug.WebSocketDebuggerUrl;
             var proxyEndpoint = $"{request.Host}{request.PathBase}/_framework/debug/ws-proxy?browser={WebUtility.UrlEncode(underlyingV8Endpoint)}";
             var devToolsUrlAbsolute = new Uri(debuggerHost + tabToDebug.DevtoolsFrontendUrl);
-            var devToolsUrlWithProxy = $"{devToolsUrlAbsolute.Scheme}://{devToolsUrlAbsolute.Authority}{devToolsUrlAbsolute.AbsolutePath}?ws={proxyEndpoint}";
+            var wsParamName = request.IsHttps ? "wss" : "ws";
```

Comment:
@Tratcher - do we need to be concerned about the case where someone is terminating HTTPS with a reverse proxy? This feature is for inner loop, so I expect a vanishingly small number of people are doing that.

Reply:
No. When using reverse proxies the guidance is to update HttpRequest accordingly so that things like IsHttps reflect the external state.
https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/proxy-load-balancer?view=aspnetcore-3.0

Reply:
@SteveSandersonMS - so just to make sure I understand the bug here:

- if the server is listening only on https
- we would send back a `ws://` url,
- chrome would try to connect over `ws://`
- the server would reject that

I'm wondering if we're missing a feature where `ws://` gets redirected to `wss://` or if potentially we have that feature, but chrome isn't following redirects.

Reply:
Either way I'm sure this is an improvement so signing off.

Reply:
> I'm wondering if we're missing a feature where `ws://` gets redirected to `wss://` or if potentially we have that feature, but chrome isn't following redirects.

Since `ws://` is never sent to the server, no, there's no websocket specific server feature to redirect it. It would use the normal http->https redirect features. You're right that the client may or may not follow said redirects for websockets.

Reply:
@rynowak Correct. We had hard coded the param name as ws regardless, then the browser would try to use a ws:// URL with the same port number as the original request. This would fail because even if the server was also listening for HTTPS, that would be on some other port number or host which we don’t know. 

If we switch to wss instead, that should match up with the hostname and port number of the current incoming request. 

I don’t know whether the client would follow any redirection in this specific case if we sent one. But the fix here eliminates the need for that. At least that’s the intention.